add container-ecosystem-mesos-openstack

rose-coste opened the pull request:
this is in response to #144 
this PR begins with editorial review, replacing PR #148 which records technical review feedback from before the repo was public; closing #148 next

stephamon commented on _best-practices/container-ecosystem-mesos-openstack.md:
```diff
@@ -0,0 +1,91 @@
+---
+
+*Mesos and OpenStack can run separately or together. OpenStack
+splits a cluster across several virtual machines; Mesos can run on bare metal or on Openstack; Mesos combines all resources, whether virtual machines or bare metal,
```
...several virtual machines. Mesos can run on bare metal or on OpenStack, and combines all resouces...

stephamon commented on _best-practices/container-ecosystem-mesos-openstack.md:
```diff
@@ -0,0 +1,91 @@
+
+Mesos' recent support of Docker containers and the development of
+Mesosphere to orchestrate containers on top of Mesos have helped Mesos
+increase in popularity. However, there
```
increase in popularity among the technical community?

stephamon commented on _best-practices/container-ecosystem-mesos-openstack.md:
```diff
@@ -0,0 +1,91 @@
+Mesos' recent support of Docker containers and the development of
+Mesosphere to orchestrate containers on top of Mesos have helped Mesos
+increase in popularity. However, there
+is confusion about how Mesos compares to other methods of
```
However, there are some questions about how Mesos compares...

stephamon commented on _best-practices/container-ecosystem-mesos-openstack.md:
```diff
@@ -0,0 +1,91 @@
+
+People tend to think of Mesos and OpenStack as competitors.
+However, Mesos and OpenStack are so different that it isn't completely practical to
+compare them. In fact, one can run Mesos on top of OpenStack [(1)](#resources); this tends
```
In fact, running Mesos on top of OpenStack is the most common operating model, but you can also run Mesos directly...

stephamon commented on _best-practices/container-ecosystem-mesos-openstack.md:
```diff
@@ -0,0 +1,91 @@
+Therefore, you can think of Mesos as one
+very large machine on which to run your applications.
+
+The promoters of Mesos argue that virtual machines were meant
```
Promoters of Mesos argue...

stephamon commented on _best-practices/container-ecosystem-mesos-openstack.md:
```diff
@@ -0,0 +1,91 @@
+The promoters of Mesos argue that virtual machines were meant
+to solve the issue of
+consolidating several differentiated workloads on far fewer machines
+than they once needed, ultimately saving money,
```
...far fewer machines, ultimately saving money...

stephamon commented on _best-practices/container-ecosystem-mesos-openstack.md:
```diff
@@ -0,0 +1,91 @@
+consolidating several differentiated workloads on far fewer machines
+than they once needed, ultimately saving money,
+simplifying management of resources,
+and moving away from traditional virtualization which was plagued with
```
...traditional virtualization, which was plagued...

stephamon commented on _best-practices/container-ecosystem-mesos-openstack.md:
```diff
@@ -0,0 +1,91 @@
+simplifying management of resources,
+and moving away from traditional virtualization which was plagued with
+horrible turnarounds and bottlenecks. This much is true. In contrast,
+Mesos to a degree reverses this pattern at the cost of removing full
```
Mesos reverses this pattern to a degree, at the cost of...

stephamon commented on _best-practices/container-ecosystem-mesos-openstack.md:
```diff
@@ -0,0 +1,91 @@
+and moving away from traditional virtualization which was plagued with
+horrible turnarounds and bottlenecks. This much is true. In contrast,
+Mesos to a degree reverses this pattern at the cost of removing full
+isolation for your application due to the operational differences between Linux containers
```
...application because of the operational differences...

stephamon commented on _best-practices/container-ecosystem-mesos-openstack.md:
```diff
@@ -0,0 +1,91 @@
+isolation for your application due to the operational differences between Linux containers
+and virtual machines.
+
+In particular, recent attention to Mesos in the Docker ecosystem centers
```
Take out "In particular" so: "Recent attention to Mesos in the Docker ecosystem..."

stephamon commented on _best-practices/container-ecosystem-mesos-openstack.md:
```diff
@@ -0,0 +1,91 @@
+A key consideration to investigate is the degree to which your
+your datacenter, stack, and application must be decoupled
+to accommodate workloads that may be in virtual machines, containers, or both.
+Timothy Prickett Morgan	suggests that OpenStack may seem most relevant to those who are most focused on datacenter operation while Mesos and Kubernetes may appeal most to those most focused on application development:
```
Make sure that extra whitespace between "Morgan" and "suggests" doesn't render

stephamon commented on _best-practices/container-ecosystem-mesos-openstack.md:
```diff
@@ -0,0 +1,91 @@
+to accommodate workloads that may be in virtual machines, containers, or both.
+Timothy Prickett Morgan	suggests that OpenStack may seem most relevant to those who are most focused on datacenter operation while Mesos and Kubernetes may appeal most to those most focused on application development:
+
+> OpenStack is gaining traction in the datacenter and Mesos and Kubernetes, while in their relative infancy, are drawing attention to the higher layers of the software stack.\[...\] It would be natural enough to think that OpenStack would end up being the center of gravity for all control functions, at least for those parts of the datacenter where open source software is preferred.\[...\] When Op ...
```
OpenStack is gaining traction in the datacenter, and Mesos and Kubernetes...

stephamon commented:
Hi @rose-coste, I've added some line comments. Let me know if you have any questions or comments!

rose-coste commented:
@stephamon, I addressed your suggestions; let me know what you think. 
After that, I think we're back to normal now so a Merge is going to work properly again. 
Thanks.

stephamon commented:
@rose-coste looks good to me :+1:
